woocommerce/woocommerce-android
# [Tablet M2] Add Margins to Details View in Order Details

Closes: #10976 


### Description

Leverages concepts utilized in https://github.com/woocommerce/woocommerce-android/pull/10949 to implement margins for all detail views within Order Details. 

- It implement details navigation controller logic.
- Let the details navigation controller keep its progress by not recreating it
- Order Details transition to a single pane for smaller screens. 


### Testing instructions

1. Open Orders on a tablet. 
2. Open an Order. 
3. Click on details such as Billing Address, Shipping Address, Customer Note and notice the margin. 
4. Go to the Products and ensure the details logic there isn't broken. 



### Images/gif

[order-details-margins_rotated.webm](https://github.com/woocommerce/woocommerce-android/assets/1509205/6d9a843d-a1be-47ad-945b-63a055477037)
